### PR TITLE
Update detect, libav, documentation, and reorganise section to improve flow

### PR DESCRIPTION
**Pull request by nathan-contino**
Bringing together all of the other camera software updates into a nice new flow. Making sure that structure, wording, linkages, etc. are all consistent across the first few sections. Demoted some unimportant sections to subsections, lifted a couple of new sections up for consistency.

### CODE REVIEW

File: documentation/asciidoc/computers/camera/rpicam_apps_intro.adoc [hunk 1]
```diff
@@ -20,9 +18,11 @@ For this reason Raspberry Pi supplies a small set of example `rpicam-apps`. Thes
 * `rpicam-raw`: Captures raw (unprocessed Bayer) frames directly from the sensor.
 * `rpicam-detect`: Not built by default, but users can build it if they have TensorFlow Lite installed on their Raspberry Pi. Captures JPEG images when certain objects are detected.
 
-Users can create their own rpicam-based applications with custom functionality to suit their own requirements. The https://github.com/raspberrypi/rpicam-apps[`rpicam-apps` source code] is freely available under a BSD-2-Clause licence.
+Recent versions of Raspberry Pi OS include the five basic `rpicam-apps`, so you can record images and videos using a camera even on a fresh Raspberry Pi OS installation.
```
Comment (lurch) [inline documentation/asciidoc/computers/camera/rpicam_apps_intro.adoc:21]: "...include the five basic `rpicam-apps` pre-installed, so..." ?

Comment (nathan-contino) [reply documentation/asciidoc/computers/camera/rpicam_apps_intro.adoc:21]: doesn't "include" render "pre-installed" redundant? I could also replace "include" with "pre-install" but that seems vaguely less correct.

File: documentation/asciidoc/computers/camera/rpicam_jpeg.adoc [hunk 2]
```diff
@@ -1,50 +1,19 @@
 === `rpicam-jpeg`
 
-`rpicam-jpeg` is a simple still image capture application. It deliberately avoids some of the additional features of `rpicam-still`, which attempts to emulate `raspistill` more fully. This means that the code is significantly easier to understand, while still providing many of the same features.
+`rpicam-jpeg` helps you capture images on Raspberry Pi devices.
```
Comment (lurch) [inline documentation/asciidoc/computers/camera/rpicam_jpeg.adoc:3]: With this introductory paragraph slimmed down so much, I guess a user wouldn't know whether to use `raspicam-still` or `raspicam-jpeg` ? :thinking:

Comment (nathan-contino) [reply documentation/asciidoc/computers/camera/rpicam_jpeg.adoc:3]: `rpicam-still` still keeps this information. But I think `raspi` has been dead long enough that we don't have to clog up our descriptions with constant `raspi` mentions. After all, we have all the libcamera-apps cruft now instead 😉

Comment (lurch) [reply documentation/asciidoc/computers/camera/rpicam_jpeg.adoc:3]: I meant more the fact that `rpicam-jpeg` is a deliberately-simple demo application, designed for code-readability; whereas `rpicam-still` is a more fully-featured application.

**Pull request merged**